src/actions.ts

```diff
@@ -333,6 +333,29 @@ export function getActions(instance: PtzOpticsInstance): CompanionActionDefiniti
 				void instance.sendCommand(PresetRecall, event.options)
 			},
 		},
+		[PtzOpticsActionId.RecallPresetFromVar]: {
+			name: 'Recall Preset From Variable',
+			options: [
+				{
+					type: 'textinput',
+					label: 'Preset Number from Variable',
+					id: 'recallPresetVariableVal',
+					useVariables: true,
+					tooltip: 'Preset number range of 0-89, 100-254',
+					default: '0',
+				},
+			],
+			callback: async (event: CompanionActionEvent) => {
+				const varPreset = await instance.parseVariablesInString(event.options.recallPresetVariableVal as string)
+				const hexval = PresetRecallOption.valmap.get(parseInt(varPreset, 10))
```

Comment:
Add a

```typescript
export function twoDigitHex(n: number): string {
    return ('0' + n.toString(16)).slice(-2)
}
```

in `options.js` (and use it in the defining of `PRESET_CHOICES` so the same conversion technique is defined only one place), then do

```typescript
    const interpolated = await instance.parseVariablesInString(String(event.options.recallPresetVariableVal))
    const preset = parseInt(interpolated, 10)
    if (preset === NaN) {
        instance.log('error', `Invalid recall preset value: ${interpolated}`)
        return
    }
    void instance.sendCommand(PresetRecall, { val: twoDigitHex(preset) })
```

Reply:
The only reason I used the map was to exclude that range in the middle that the drop-down doesn't support.  What do you think, similar to the first function, about having a common function that determines if the preset is in the "valid" range (excludes 90-100)?  That function could be called in the loop as well to exclude from the dropdown and to invalidate any variables?

Reply:
I ended up adding an isValid to cover that as I mentioned